### ISSUE TITLE
Allow to choose between modin[ray] and modin[dask]

### ISSUE THREAD

**Issue by szborows** [jmcarpenter2/swifter#150]
Dependency `modin[ray]` is hardcoded into `setup.py` and it conflicts with tensorflow's model analysis package on `pyarrow`:

```
pyarrow==1.0 (from modin[ray]==0.9.1->swifter==1.0.7...
pyarrow<0.18,>=0.17 (from tensorflow-model-analysis==0.26.0->model-card-toolkit==0.1.2...
```

It would be great if one could choose to install `modin[dask]` instead of ray version.

Maybe the first step would be to make modin dependency optional as requested in #147 and then let people decide which version should be installed.

**Comment by lukleu** [comment]
+1

**Comment by jmcarpenter2** [comment]
Hey @szborows this is a good call! I will work on this shortly, thanks for flagging!

**Comment by jmcarpenter2** [comment]
Resolved in `swifter==1.0.8` #156 

Please refer to the README on how to install swifter with modin[dask] (`pip install swifter[modin-dask]`)
